hamocc/mo_write_netcdf_var.F90

```diff
@@ -0,0 +1,211 @@
+
+  subroutine write_netcdf_var(ncid,desc,arr,klev,time)
+
+    !**************************************************************************
+    ! Gathers a global variable from all PEs and writes it to a NETCDF file
+    ! The NETCDF File is only accessed by mnproc=1
+    !**************************************************************************
+
+    use netcdf,  only: nf90_noerr,nf90_inq_varid,nf90_strerror,nf90_put_var
+    use mod_xc,  only: itdm,jtdm,jdm,lp,mnproc,nbdy,idm,xchalt,xcaget
+    use mod_dia, only: iotype
+#ifdef PNETCDF
+    use mod_xc,  only: i0,ii,jj,j0,mproc,mpe_1,nproc,xcgetrow
+#endif
+#ifdef PNETCDF
+#  include <pnetcdf.inc>
+#  include <mpif.h>
+#endif
+
+    ! Arguments
+    integer,          intent(in)  :: ncid
+    character(len=*), intent(in)  :: desc
+    integer,          intent(in)  :: klev
+    integer,          intent(in)  :: time
+    real,             intent(out) :: arr(idm,jdm,klev)
+
+    ! Local variables
+    integer :: k,i,j
+    integer :: ndims
+    real    :: arr_g(itdm,jtdm)
+    integer :: ncstat
+    integer :: ncvarid
+    integer, allocatable :: start(:),count(:)
+    real,    allocatable :: arr_g1(:,:,:)
+    real,    allocatable :: arr_l(:,:,:)
+#ifdef PNETCDF
+    integer (kind=MPI_OFFSET_KIND), allocatable :: istart(:),icount(:)
+#endif
+
+    ! Write NETCDF data
+
+    if (klev.eq.1.and.time.eq.0) then
+      ndims=2
+    elseif (klev.eq.1.or.time.eq.0) then
+      ndims=3
+    else
+      ndims=4
+    endif
+    if (IOTYPE==0) then
+
+      allocate(start(ndims))
+      allocate(count(ndims))
+      allocate(arr_l(1-nbdy:idm+nbdy,1-nbdy:jdm+nbdy,1))
+      arr_l=0.0
+      start(1)=1
+      count(1)=itdm
+      start(2)=1
+      count(2)=jtdm
+      if (klev.gt.1.or.time.gt.0) then
+        if (klev.gt.1.and.time.gt.0) then
+          start(3)=1
+          count(3)=1
+          start(4)=time
+          count(4)=1
+        else if (klev.gt.1.and.time.eq.0) then
+          start(3)=1
+          count(3)=1
+        else
+          start(3)=time
+          count(3)=1
+        endif
+      endif
+
+      do k=1,klev
+        do j=1,jdm
+          do i=1,idm
+            arr_l(i,j,1)=arr(i,j,k)
+          enddo
+        enddo
+        call xcaget(arr_g,arr_l,1)
+        if (mnproc.eq.1) then
+          if (k.gt.1) then
+            start(3)=k
+            count(3)=1
+          endif
+
+          ncstat=nf90_inq_varid(ncid,desc,ncvarid)
+          if (ncstat.ne.nf90_noerr) then
+            write(lp,'(4a)') 'nf90_inq_varid: ',trim(desc),': ',nf90_strerror(ncstat)
+            call xchalt('(write_netcdf_var)')
+            stop '(write_netcdf_var)'
+          endif
+
+          ncstat=nf90_put_var(ncid,ncvarid,arr_g,start,count)
+          if (ncstat.ne.nf90_noerr) then
+            write(lp,'(4a)') 'nf90_put_var: ',trim(desc),': ',nf90_strerror(ncstat)
+            call xchalt('(write_netcdf_var)')
+            stop '(write_netcdf_var)'
+          endif
+
+          ! ncstat=nf90_sync(ncid)
```

Comment:
Why commented out? - more a question.

Reply:
I'm not sure why this was commented out. @JorgSchwinger @TomasTorsvik - do you know?

Reply:
Sorry, my bad. I guess, I started to get tired - I overlooked the comment marks at the beginning of the line in the original code. Same applies for below.